### PR TITLE
Add Standalone Vue Remote Devtools App (electron shell) (#451)

### PR DESCRIPTION
**Pull request by rigor789**
This is a WIP for adding an electron shell and remote debugging to VueDevtools.

Discussion is at #451

### CODE REVIEW

**Comment by crswll** [comment]
Might be able to pick up some tips from this: https://github.com/vuetwo/vuetron

**Comment by michalsnik** [comment]
@rigor789 I updated your PR on this branch https://github.com/vuejs/vue-devtools/tree/add-remote-devtools take a look.

Quick instruction:
1. Go to `shells/electron` and run:
```
npm run dev
npm start
```

2. Inject script on page that you want to inspect in `HEAD`:
```
<script src="http://localhost:8098"></script>
```

3. Refresh page and you should be able to properly inspect it :)

File: src/backend/hook.js [hunk 1]
```diff
@@ -12,75 +12,77 @@
 export function installHook (window) {
   let listeners = {}
 
-  const hook = {
-    Vue: null,
+  if (!window.hasOwnProperty('__VUE_DEVTOOLS_GLOBAL_HOOK__')) {
```
Comment (michalsnik) [inline src/backend/hook.js]: Perhaps without extra indentation?
```
if (window.hasOwnProperty('__VUE_DEVTOOLS_GLOBAL_HOOK__')) return
```

File: shells/dev/target-electron.html [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+<html lang="en">
+  <head>
+    <meta charset="utf-8">
+    <title></title>
+  </head>
+  <body>
+    <div id="app"></div>
+    <div id="shadow"></div>
+    <script src="http://localhost:8098"></script>
+    <script src="build/hook.js"></script>
```
Comment (michalsnik) [inline shells/dev/target-electron.html]: Hook is not needed here, it's being loaded by above script

**Pull request merged**